pkg/kubelet/kubelet.go

```diff
@@ -1469,8 +1471,39 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	podStatus := o.podStatus
 	updateType := o.updateType
 
+	//Get the hostname for traces
+	activeHost, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("could not retrieve hostname")
+	}
+
+	// Create and register an OpenCensus
+	// Stackdriver Trace exporter.
+	exporter, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: "samnaser-gke-dev-217421",
+	})
+
+	if err != nil {
+		return fmt.Errorf("error configuring stackdriver opencensus exporter: %v", err)
+	}
+
+	// Register our exporter with our tracer and set a liberal sampling policy for testing
+	trace.RegisterExporter(exporter)
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+
+	// Parent span, annotate it with the hostname
+	ctx := context.Background()
+	ctx, span := trace.StartSpan(ctx, "Kubelet: SyncPod")
+	span.AddAttributes(trace.StringAttribute("host", activeHost))
+	defer span.End()
+
+	podKillCtx, podKillSpan := trace.StartSpan(ctx, "Pod kill logic")
```

Comment:
instead of producing a new context, can we overrwrite the `ctx` variable each time?  That would make the code less brittle, and easier to add new spans if we need to in the future.